library/ssl_tls.c

```diff
@@ -5233,7 +5341,13 @@ static int ssl_check_dtls_clihlo_cookie(
 
     (void)mbedtls_platform_put_uint16_be( &obuf[11], ( *olen - 13 ) );
 
-    return( MBEDTLS_ERR_SSL_HELLO_VERIFY_REQUIRED );
+    if( cli_id_dup == cli_id && cli_id_len_dup == cli_id_len &&
+        in_dup == in && in_len_dup == in_len && obuf_dup == obuf &&
+        buf_len_dup == buf_len && olen_dup == olen )
+    {
+        return( MBEDTLS_ERR_SSL_HELLO_VERIFY_REQUIRED );
+    }
+    return ( MBEDTLS_ERR_PLATFORM_FAULT_DETECTED );
```

Comment:
What is the meaning of verifying `MBEDTLS_ERR_PLATFORM_FAULT_DETECTED` before returning other error code? Is this really necessary?

Reply:
It's not a library error _per se_, it means that a hello verification is requested - it's part of the protocol. So in this case, it's considered a normal part of operation.